src/Artsy/Router/Utils/Route.tsx

```diff
@@ -28,33 +27,8 @@ function createRender({
 }: CreateRenderProps) {
   return (renderArgs: RenderArgProps) => {
     const { Component, props, error } = renderArgs
-    let status: number
-    let message: string
-    if (error) {
-      if (error instanceof HttpError) {
-        throw error
-      } else if (error.name === "RRNLRequestError") {
-        // TODO: Better error typing.
-        // @ts-ignore
-        const firstError = get(error, e => e.res.errors[0])
-        const statusCodes = get(
-          firstError,
-          e => e.extensions.httpStatusCodes,
-          []
-        )
-        if (statusCodes.length === 1) {
-          status = statusCodes[0]
-          message = firstError.message
-        }
-      } else {
-        status = 500
-        message = error.message
-      }
-
-      // TODO: Need upstream fix type in found as it complains about missing
-      // second argument.
-      // @ts-ignore
-      throw new HttpError(status, message)
+    if (error && error instanceof HttpError) {
+      throw error
```

Comment:
🎉

Reply:
Note that we need 
```
throw new HttpError(status, message)
```
For errors to be redirected to the error page within the router: https://github.com/4Catalyzer/found#error-handling

Reply:
This `error` is already an `HttpError` with the appropriate status, from https://github.com/artsy/reaction/blob/c7d112dc3fc53da07c191ebbc1cf6e7fb01fd48d/src/Artsy/Relay/createRelaySSREnvironment.ts#L115 (`httpError` is probably not a great variable name, that is the status code. We don't pass in the `message`).

So this all should work (ie-  a principal field 404 by going to a non existent artist/artwork page correctly returns a 404 and renders that error page).